engine/src/main/java/org/terasology/rendering/cameras/Camera.java

```diff
@@ -73,6 +76,18 @@
     /* (Water) Reflection */
     private boolean reflected;
     private float reflectionHeight = 32;
+    
+    /* Used for Underwater Checks */
+    private WorldProvider worldProvider;
+    private RenderingConfig renderingConfig;
+    
+    public Camera(){
+    }
+    
+    public Camera(WorldProvider worldProvider, RenderingConfig renderingConfig){
```

Comment:
My concern here is that, as you rightly noticed, there are cameras that are not going underwater, i.e. the ShadowMapCamera.

One way to handle this would be to make a SubmersibleCamera class extending the Camera class and adding only the underwater check functionality. PerspectiveCamera and the OpenVRCamera would then extend that rather than the plain Camera class. If we look at the way the code works now, this would be enough.

We could use more sophisticated methods, so that cameras can acquire the underwater check functionality at runtime rather than have it statically compiled in. But I think for the time being let's keep it simple. Cameras are slated to go through some major changes anyway (see #2797) and it is best if we don't invest toooo much time on them.